lib/optimist.rb

```diff
@@ -316,10 +331,14 @@ def parse(cmdline = ARGV)
         else                       raise CommandlineError, "invalid argument syntax: '#{arg}'"
       end
 
-      sym = nil if arg =~ /--no-/ # explicitly invalidate --no-no- arguments
+      if arg =~ /--no-/ # explicitly invalidate --no-no- arguments
```

Comment:
Do we want to check if the string starts with `--no-` or just includes `--no-`?  The regex right now is the latter, but I don 't think that's what we want.

Regardless, I don't see us using any regex special chars nor the result from this, so I think we could use `.start_with?` or `.include?` respectively, with a string `"--no-"`

Reply:
hrm, interesting. Agree and we can do this as part of this PR or another.